b2share/modules/records/triggers.py

```diff
@@ -60,12 +70,18 @@ def set_record_schema(record, **kwargs):
     record['$schema'] = community_schema_json_schema_link(schema)
 
 
+def set_record_owner(record, **kwargs):
+    if not current_user.is_authenticated:
```

Comment:
It think that it would be better to leave the `is_authenticated` check to the permission factory. See https://github.com/inveniosoftware/invenio-records-rest/blob/144dee816ee05c10e09e72eaf2f5b18c74285977/invenio_records_rest/views.py#L264

The permission factory just has to return a permission which returns False when `can()` is called.